command/project/add.go

```diff
@@ -135,6 +135,12 @@ func (p *Add) Store(e interface{}) bind.Status {
 
 func (p *Add) Exec() error {
 	proj := p.Project()
+	if finfo, err := os.Stat(proj.Path); (err == nil && !finfo.IsDir()) || (err != nil && !os.IsNotExist(err)) {
+		return fmt.Errorf("You can't choose file %s as path for project", proj.Path)
+	}
+	if finfo, err := os.Stat(proj.Gopath); (err == nil && !finfo.IsDir()) || (err != nil && !os.IsNotExist(err)) {
+		return fmt.Errorf("You can't choose file %s as gopath for project", proj.Gopath)
+	}
```

Comment:
These `if` statements also are probably better served in a separate function - maybe `isDir(proj.Path)` and `isDir(proj.Gopath)` or something?  the boolean expressions are just a bit too complicated to read easily in their current state.

Reply:
Agree. I will move it to separate function, split condition and also will write comments. But I think isCorrectPath would be better as name because not-exist path not a dir nor a file but correct path.

Reply:
Works for me :)